java/generate-offline-map-with-local-basemap/src/main/java/com/esri/arcgisruntime/sample/generateofflinemapwithlocalbasemap/ProgressDialogFragment.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.generateofflinemapwithlocalbasemap;
+
+import android.app.Dialog;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.DialogFragment;
+
+public class ProgressDialogFragment extends DialogFragment {
+
+  private static final String ARGS_TITLE = ProgressDialogFragment.class.getSimpleName() + "_title";
+  private static final String ARGS_MESSAGE = ProgressDialogFragment.class.getSimpleName() + "_message";
+  private static final String ARGS_CANCEL = ProgressDialogFragment.class.getSimpleName() + "_cancel";
+
+  private OnProgressDialogDismissListener mOnProgressDialogDismissListener;
+
+  private String title;
+  private String message;
+  private String cancel;
+
+  public static ProgressDialogFragment newInstance(String title, String message, String cancel) {
+    ProgressDialogFragment fragment = new ProgressDialogFragment();
+    Bundle args = new Bundle();
+    args.putString(ARGS_TITLE, title);
+    args.putString(ARGS_MESSAGE, message);
+    args.putString(ARGS_CANCEL, cancel);
+    fragment.setArguments(args);
+    return fragment;
+  }
+
+  @Override public void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    // prevent re-creation during configuration chance to allow us to dismiss this DialogFragment
+    setRetainInstance(true);
```

Comment:
I'm starting to question whether this is a good idea. Generally we shouldn't use it but I understand I left it in my previous PR. Will investigate.

Reply:
I've done a refactor of the Analyze Hotspots ticket again and move quite alot around to attempt to retain state over configuration changes. Happy to leave this here and we can discuss this further.

Reply:
I actually kind of like the setRetainInstance(true) code, even if it involved a work around of an Android API bug...